ci/linkcheck.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+if [ "$TRAVIS_EVENT_TYPE" = "cron" ] ; then # running in cron job
+  FLAGS=""
+
+  echo "Doing full link check."
+elif [ "$CI" = "true" ] ; then # running in PR CI build
+  CHANGED_FILES=$(git diff --name-only $TRAVIS_COMMIT_RANGE | tr '\n' ' ')
```

Comment:
> TRAVIS_COMMIT_RANGE: The range of commits that were included in the push or pull request. (Note that this is empty for builds triggered by the initial commit of a new branch.)

After reading over the docs I think this means for 'push' runs, not 'pr' runs, so this is probably fine, but can you add an assert just in case?
```suggestion
  if ! [ "$TRAVIS_EVENT_TYPE" = push ] || [ -n "$TRAVIS_COMMIT_RANGE" ]; then
    echo "error: unexpected state: COMMIT_RANGE must be non-empty in CI"
    exit 1
  fi
  CHANGED_FILES=$(git diff --name-only $TRAVIS_COMMIT_RANGE | tr '\n' ' ')
```

Reply:
I'm not sure I understand the concern... wouldn't `[ -n "$TRAVIS_COMMIT_RANGE" ]` be sufficient?

Reply:
No, because it's expected that this will be empty on the first push in a new branch. So people will get spurious emails about the PR failing even though nothing went wrong.

Reply:
Hmm... As I understand it, neither the range nor the environment variable will be empty for push or pr runs... that is, if you either create a new PR or push commits to a PR, TRAVIS_COMMIT_RANGE will exist and contain a non-empty range with all changes back to the merge-base with master...

Thus, I believe the `! [ "$TRAVIS_EVENT_TYPE" = push ]` is incorrect?

Reply:
I'm not _sure_ that's what the docs say, but this will fail noisily if it fails so I'm fine with removing the `push` check.

Reply:
Do we even need to use `$TRAVIS_COMMIT_RANGE`? Can't we just use `git diff --name-only master...` like for running locally?

Reply:
@camelid no, because we have no guarentee `master` on travis is up-to-date with master on github. We don't even know if the remote is `rust-lang/` or the contributors fork (I think normally the fork?)

Reply:
Ok, I have added the check.